### PR TITLE
Project Proposal: EIP-8304 Trustless Log Index

### PR DESCRIPTION
**Pull request by Dyslex7c**
This PR adds my project proposal on Trustless Tx and Log Index. 

I've added two code snippets to demonstrate a rough overview how implementation of index entry generation and table root calculation would look like using existing primitives in the Nethermind codebase, can remove it if it seems extra.

### CODE REVIEW

File: projects/nethermind-eip-8304-trustless-log-index.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+- Write a Hive simulator test suite covering: correct root generation across all levels, reorg handling, sync recovery, restart resilience, malformed proof rejection, boundary conditions (empty blocks, max-entry blocks, ring buffer wraparound, genesis edge cases).
+- Performance benchmarking: measure per-block overhead of index generation and system call, disk growth rate, proof generation latency, and proof sizes at each table level.
+- Prepare the ppt presentation for showcase.
+
+## Possible Challenges
+
+Specification instability: EIP-8304 is a Draft. The contract address, deployment transaction, and encoding details are not finalized. Entry encoding lengths in the spec text differ between sections (the "entry types" table shows `4+32+8 = 44` bytes for block entries while the binary examples use a 2-byte type prefix giving 42 bytes). Mitigation strategy: pin to a specific spec commit, maintain the reference root calculator as the authoritative encoding oracle, and keep the encoding layer isolated behind an interface so changes propagate cleanly.
```
Comment (taxmeifyoucan) [inline projects/nethermind-eip-8304-trustless-log-index.md:256]: other 8304 implementations proposed in the cohort, geth and reth, use different values for encoding lengths.. would be best to coordinate on this. Also to elaborate how is the coordination happening between fellows

File: projects/nethermind-eip-8304-trustless-log-index.md [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+}
+```
+
+### Multi-Level Table Merging
+
+The EIP mandates five table levels with sizes `[1, 4, 16, 64, 256]`. Higher-level tables are produced by k-way merging the sorted entries of their constituent lower-level tables. Since entries are already sorted, the merge is a standard sorted-merge operation.
+
+Higher-level tables are published with a delay (`TABLE_SIZES[i]/4` blocks) to allow asynchronous background merging without adding latency to block processing. The merge scheduler must handle:
+
+- Deadlines: Level-i merges must complete within `TABLE_SIZES[i]/4` blocks.
```
Comment (taxmeifyoucan) [inline projects/nethermind-eip-8304-trustless-log-index.md:148]: The performance is a big questions here and I am wondering what would it take to do merging of certain depth on resource constrained hardware

File: projects/nethermind-eip-8304-trustless-log-index.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+  → eth_getProof for index contract storage slot
+    → Table root (SSZ List[Hash32] root)
+      → Sorted entries + Merkle proof
+        → Matching entries with (blockNumber, txIndex, logIndex)
+          → Block entry → target block header hash
+            → Receipt proof against receiptsRoot
+              → Verified log (address, topics, data)
+```
+
+## Roadmap
```
Comment (taxmeifyoucan) [inline projects/nethermind-eip-8304-trustless-log-index.md:206]: I assume weeks in the roadmap are counting from now not since the beginning of the cohort? It's a bit confusing

**Review by taxmeifyoucan — COMMENTED** [review]
thanks for the proposal!

File: projects/nethermind-eip-8304-trustless-log-index.md (hunk 1 above)
Comment (Dyslex7c) [reply projects/nethermind-eip-8304-trustless-log-index.md:256]: sure will coordinate with them and check it. We are communicating via a signal group where we are sharing our progress and syncing with each other, I went through their proposals and drafted mine based on that so we're on the same page

File: projects/nethermind-eip-8304-trustless-log-index.md (hunk 2 above)
Comment (Dyslex7c) [reply projects/nethermind-eip-8304-trustless-log-index.md:148]: I can validate this with benchmarking profiles to see how it performs on constrained hardware

File: projects/nethermind-eip-8304-trustless-log-index.md (hunk 3 above)
Comment (Dyslex7c) [reply projects/nethermind-eip-8304-trustless-log-index.md:206]: yeah counting from now, for example week 12 to week 14 is the last phase between late October and early Nov just before project showcase. It's a bit stretched but I'll try to finish early if possible
